File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int n = cardPoints.length;
        int l=k-1 , r = n-1 , lsum=0 , rsum=0 ;
        for (int i=0 ;i <= l ; i++) lsum += cardPoints[i];
        int maxSum = lsum;
        while( r >= n-k ){
            lsum -= cardPoints[l];
            rsum += cardPoints[r];
            maxSum = Math.max( maxSum , lsum + rsum );
            l--;
            r--;
        }
        return maxSum;

    }
}

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution(object):
    def maxScore(self, cardPoints, k):
        """
        :type cardPoints: List[int]
        :type k: int
        :rtype: int
        """
        
        n = len(cardPoints)
        left_sum = sum(cardPoints[:k])
        max_sum = left_sum
        
        right_sum = 0
        
        for i in range(1, k+1):
            left_sum -= cardPoints[k - i]
            right_sum += cardPoints[n - i]
            max_sum = max(max_sum, left_sum + right_sum)
        
        return max_sum